failed to build source code 10.2

### Environment

```markdown
- **OS**: Ubuntu 22.04
- **Sage Version**: 10.2
```


### Steps To Reproduce

I met the error :

make[1]: Entering directory '/home/egger/sagemath/sage'
rm -rf config/install-sh config/compile config/config.guess config/config.sub config/missing configure build/make/Makefile-auto.in
rm -f src/doc/en/installation/*.txt
rm -rf src/doc/en/reference/spkg/*.rst
for a in environment environment-optional src/environment src/environment-dev src/environment-optional; do rm -f $a.yml $a-3.[89].yml $a-3.1[0-9].yml; done
rm -f src/Pipfile
rm -f src/pyproject.toml
rm -f src/requirements.txt
rm -f src/setup.cfg
make[1]: Leaving directory '/home/egger/sagemath/sage'
bootstrap:: installing 'm4/sage_spkg_configures.m4'
bootstrap:: installing 'm4/sage_spkg_versions.m4'
bootstrap:: installing 'm4/sage_spkg_versions_toml.m4'
./bootstrap: 1: src/bootstrap: not found
make: *** [Makefile:328: configure] Error 1


### Config log

make[1]: Entering directory '/home/egger/sagemath/sage'
rm -rf config/install-sh config/compile config/config.guess config/config.sub config/missing configure build/make/Makefile-auto.in
rm -f src/doc/en/installation/*.txt
rm -rf src/doc/en/reference/spkg/*.rst
for a in environment environment-optional src/environment src/environment-dev src/environment-optional; do rm -f $a.yml $a-3.[89].yml $a-3.1[0-9].yml; done
rm -f src/Pipfile
rm -f src/pyproject.toml
rm -f src/requirements.txt
rm -f src/setup.cfg
make[1]: Leaving directory '/home/egger/sagemath/sage'
bootstrap:: installing 'm4/sage_spkg_configures.m4'
bootstrap:: installing 'm4/sage_spkg_versions.m4'
bootstrap:: installing 'm4/sage_spkg_versions_toml.m4'
./bootstrap: 1: src/bootstrap: not found
make: *** [Makefile:328: configure] Error 1

### Package logs

_No response_

### Additional Information

_No response_

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

--- a/configure.ac
+++ b/configure.ac
@@ -166,6 +166,9 @@ You are attempting to build Sage on $host,
 which is not a supported platform for Sage]]);;
 esac
 
+dnl Issue #38039: Alert users on WSL early about a source tree
+dnl that has not been cloned correctly.
+dnl This will save users from running into build errors later.
 AC_MSG_CHECKING([the symbolic links in the source tree])
 AS_IF([test -L build/pkgs/liblzma/package-version.txt], [dnl OK
     AC_MSG_RESULT([intact])

--- a/configure.ac
+++ b/configure.ac
@@ -173,19 +173,19 @@ AC_MSG_CHECKING([the symbolic links in the source tree])
 AS_IF([test -L build/pkgs/liblzma/package-version.txt], [dnl OK
     AC_MSG_RESULT([intact])
 ], [dnl Either replaced by a copy (OK, as done in the sage-conf_pypi sdist)
-   dnl  or replaced by the name of the link target (BAD, misconfigured git clone in WSL)
-   AS_IF([test ! -e build/pkgs/sagelib/src], [dnl
-       AC_MSG_RESULT([intact (with links to embedded source trees pruned)])
-   ], [dnl
-       AS_IF([test -d build/pkgs/sagelib/src], [dnl
-           AC_MSG_RESULT([broken (links to directories have been replaced by copies)])
-       ], [dnl
-           AC_MSG_RESULT([broken])
-       ])
-       AC_MSG_NOTICE([Sorry, the symbolic links in this source tree are broken.])
-       AC_MSG_NOTICE([See README.md for installation instructions.])
-       AC_MSG_ERROR([Exiting, as Sage cannot be built from this source tree.])
-   ])
+    dnl  or replaced by the name of the link target (BAD, misconfigured git clone in WSL)
+    AS_IF([test ! -e build/pkgs/sagelib/src], [dnl
+        AC_MSG_RESULT([intact (with links to embedded source trees pruned)])
+    ], [dnl
+        AS_IF([test -d build/pkgs/sagelib/src], [dnl
+            AC_MSG_RESULT([broken (links to directories have been replaced by copies)])
+        ], [dnl
+            AC_MSG_RESULT([broken])
+        ])
+        AC_MSG_NOTICE([Sorry, the symbolic links in this source tree are broken.])
+        AC_MSG_NOTICE([See README.md for installation instructions.])
+        AC_MSG_ERROR([Exiting, as Sage cannot be built from this source tree.])
+    ])
 ])
 
 ###############################################################################

--- a/configure.ac
+++ b/configure.ac
@@ -166,6 +166,24 @@ You are attempting to build Sage on $host,
 which is not a supported platform for Sage]]);;
 esac
 
+AC_MSG_CHECKING([the symbolic links in the source tree])
+AS_IF([test -L build/pkgs/liblzma/package-version.txt], [dnl OK
+    AC_MSG_RESULT([intact])
+], [dnl Either replaced by a copy (OK, as done in the sage-conf_pypi sdist)
+   dnl  or replaced by the name of the link target (BAD, misconfigured git clone in WSL)
+   AS_IF([test ! -e build/pkgs/sagelib/src], [dnl
+       AC_MSG_RESULT([intact (with links to embedded source trees pruned)])
+   ], [dnl
+       AS_IF([test -d build/pkgs/sagelib/src], [dnl
+           AC_MSG_RESULT([broken (links to directories have been replaced by copies)])
+       ], [dnl
+           AC_MSG_RESULT([broken])
+       ])
+       AC_MSG_NOTICE([Sorry, the symbolic links in this source tree are broken.])
+       AC_MSG_NOTICE([See README.md for installation instructions.])
+       AC_MSG_ERROR([Exiting, as Sage cannot be built from this source tree.])
+   ])
+])
 
 ###############################################################################
 # Check general programs

